test/python/result/test_mitigators.py

```diff
@@ -172,112 +56,175 @@ def compare_results(self, res1, res2):
+        return Counts(dict(Counter(results)))
+
+    @staticmethod
+    def ghz_3_circuit():
+        """A 3-qubit circuit generating |000>+|111>"""
+        c = QuantumCircuit(3)
+        c.h(0)
+        c.cx(0, 1)
+        c.cx(1, 2)
+        return (c, "ghz_3_ciruit", 3)
+
+    @staticmethod
+    def first_qubit_h_3_circuit():
+        """A 3-qubit circuit generating |000>+|001>"""
+        c = QuantumCircuit(3)
+        c.h(0)
+        return (c, "first_qubit_h_3_circuit", 3)
+
+    @staticmethod
+    def assignment_matrices():
+        """A 3-qubit readout noise assignment matrices"""
+        # from FakeYorktown
+        assignment_matrices = [
+            np.array([[0.98828125, 0.04003906], [0.01171875, 0.95996094]]),
+            np.array([[0.99023438, 0.02929688], [0.00976562, 0.97070312]]),
+            np.array([[0.984375, 0.02441406], [0.015625, 0.97558594]]),
+        ]
+        return assignment_matrices
+
+    @staticmethod
+    def counts_data(circuit, assignment_matrices, shots=1024):
+        """Generates count data for the noisy and noiseless versions of the circuit simulation"""
+        full_assignment_matrix = assignment_matrices[0]
+        for m in assignment_matrices[1:]:
+            full_assignment_matrix = np.kron(full_assignment_matrix, m)
+        num_qubits = len(assignment_matrices)
+        ideal_assignment_matrix = np.eye(2**num_qubits)
+        counts_ideal = TestReadoutMitigation.simulate_circuit(
+            circuit, ideal_assignment_matrix, num_qubits, shots
+        )
+        counts_noise = TestReadoutMitigation.simulate_circuit(
+            circuit, full_assignment_matrix, num_qubits, shots
+        )
+        probs_noise = {key: value / shots for key, value in counts_noise.items()}
+        return counts_ideal, counts_noise, probs_noise
+
+    def test_mitigation_improvement(self):
+        """Test whether readout mitigation led to more accurate results"""
+        shots = 1024
+        assignment_matrices = self.assignment_matrices()
+        num_qubits = len(assignment_matrices)
+        mitigators = self.mitigators(assignment_matrices)
+        circuit, circuit_name, num_qubits = self.ghz_3_circuit()
+        counts_ideal, counts_noise, probs_noise = self.counts_data(
+            circuit, assignment_matrices, shots
+        )
+        unmitigated_error = self.compare_results(counts_ideal, counts_noise)
+        unmitigated_stddev = stddev(probs_noise, shots)
+
+        for mitigator in mitigators:
+            mitigated_quasi_probs = mitigator.quasi_probabilities(counts_noise)
+            mitigated_probs = (
+                mitigated_quasi_probs.nearest_probability_distribution().binary_probabilities(
+                    num_bits=num_qubits
                 )
+            )
+            mitigated_error = self.compare_results(counts_ideal, mitigated_probs)
+            self.assertTrue(
+                mitigated_error < unmitigated_error * 0.8,
```

Comment:
I know of no general result here. This specific number was chosen because it works with the current data, allowing for detections of changes to the mitigators which make it fail on this specific data. This was done with the hard-coded numbers in mind; I agree that by relying on getting the numbers from the fake backend we introduce fuzziness.